modules/overrides/config.go

```diff
@@ -200,7 +206,8 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	legacyCfg.UserConfigurableOverridesConfig = c.UserConfigurableOverridesConfig
 
 	if err := unmarshal(&legacyCfg); err != nil {
-		return err
+		// unmarshaling failed twice, return the first error
+		return originalError
```

Comment:
I'm not sure if we should combine both errors instead here? I'm worried that might be too confusing.